rando/src/main.rs

```diff
@@ -45,19 +78,21 @@ fn shuffle_area(
                     let chest = &chest_table[id as usize];
 
                     // Ensure it is not the medallion.
-                    if chest.item_id < 0x12 || chest.item_id >= (0x12 + 8) {
-                        chest_ids.push(id);
-                        chest_contents.push(chest.clone());
+                    if chest.item_id < 0x10 || chest.item_id >= (0x12 + 8) {
+                        if let Some(randomizable_chest) = randomizable_chests.pop() {
+                            chest_ids.push(id);
+                            chest_contents.push(randomizable_chest);
+                        } else {
+                            return Err(format_err!("Terrible error, ran out of chests! T_T"));
+                        }
                     }
                 }
             }
         }
     }
 
-    // Next shuffle the chest contents.
     chest_contents.shuffle(rng);
```

Comment:
I think this shuffle is unneeded because `randomizable_chests` is shuffled in main.

Reply:
Yeah, it should've been the new chest table